espresso-support/core/src/main/java/com/novoda/espresso/AccessibilityViewMatchers.java

```diff
@@ -0,0 +1,187 @@
+package com.novoda.espresso;
+
+import android.content.res.Resources;
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+import android.support.annotation.StringRes;
+import android.view.View;
+import android.view.accessibility.AccessibilityNodeInfo;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import static org.hamcrest.Matchers.is;
+
+public class AccessibilityViewMatchers {
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    public static Matcher<? super View> withUsageHintOnClick(@StringRes final int resourceId) {
+        return new TypeSafeMatcher<View>() {
+
+            private String resourceName = null;
+            private String expectedText = null;
+
+            @Override
+            protected boolean matchesSafely(View view) {
+                if (!view.isClickable()) {
+                    return false;
+                }
+
+                if (null == this.expectedText) {
+                    try {
+                        expectedText = view.getResources().getString(resourceId);
+                        resourceName = view.getResources().getResourceEntryName(resourceId);
+                    } catch (Resources.NotFoundException ignored) {
+                        // view could be from a context unaware of the resource id.
```

Comment:
Simplified these by getting rid of the catch - if the resources aren't found, the test will fail (which this did) and I think having the exception will be a better description for why the test failed!

commit 18e6d202509d51d82aca202ab2ad65da3aa20073
Author: Ataul Munim <hello@ataulm.com>
Date:   Tue Mar 28 13:55:09 2017 +0100

    Remove try/catch from ViewMatchers that read from resources
    
    - ResourceNotFound is legit enough to make the test fail and understand why